File: Leetcode/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.py
class Solution:
    def isPrefixOfWord(self, sen: str, sW: str) -> int:
        wordList = []
        temp = ""
        m = len(sW)
        for i in range(len(sen)):
            if sen[i] != " ":
                temp += sen[i]
            else:
                wordList.append(temp)
                temp = ""
        wordList.append(temp)

        for i in range(len(wordList)):
            n = 0
            arr = ""
            while n < m and len(wordList[i]) >= m:
                if wordList[i][n] == sW[n]:
                    arr += wordList[i][n]
                    print(wordList[i][n], "==" , sW[n])
                    print(arr)
                    n+=1
                    if arr == sW:
                        return i+1
                else:
                    arr = ""
                    break
        return -1

File: Leetcode/1566-check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence/check-if-a-word-occurs-as-a-prefix-of-any-word-in-a-sentence.cpp
class Solution {
public:
    int isPrefixOfWord(string sentence, string searchWord) {
        vector<string> words;
        string temp;
        for(char ch : sentence) {
            if(ch != ' ') temp += ch;
            else { words.push_back(temp); temp.clear(); }
        }
        words.push_back(temp);

        int m = searchWord.size();
        for(int i = 0; i < words.size(); i++) {
            if(words[i].size() >= m && words[i].substr(0, m) == searchWord)
                return i + 1;
        }
        return -1;
    }
};
